web/settings/context.go

```diff
@@ -47,6 +54,40 @@ func onboarded(c echo.Context) error {
 		}
 	}
 	redirect := i.OnboardedRedirection().String()
+
+	// Redirect to permissions screen if we are in a mobile onboarding
+	if i.OnboardingSecret != "" {
+		// Generate the deeplink
+		deeplink := fmt.Sprintf("cozy%s://%s", i.OnboardingApp, i.Domain)
+
+		// Generate the OAuth client
+		OAuthClient := oauth.Client{
+			RedirectURIs: []string{deeplink},
+			ClientName:   i.OnboardingApp,
+			SoftwareID:   "github.com/cozy/cozy-stack",
+		}
+		OAuthClient.Create(i)
+		i.OnboardingClientID = OAuthClient.ClientID
+
+		// Redirection
+		// Generate state
+		b := make([]byte, 32)
```

Comment:
I've updated the sequence diagram: it should be the webview that creates this state. And the state should be checked by the webview after the redirection (ie the state the webview has generated == the state in the query string in the redirected URL)